services/network/client/infrastructure/input/tcp-check.controller.go

```diff
@@ -0,0 +1,108 @@
+/* Copyright (c) 2022 Snowflake Inc. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+*/
+
+package input
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	pb "github.com/Snowflake-Labs/sansshell/services/network"
+	app "github.com/Snowflake-Labs/sansshell/services/network/client/application"
+	"github.com/Snowflake-Labs/sansshell/services/util"
+	cliUtils "github.com/Snowflake-Labs/sansshell/services/util/cli"
+	"github.com/Snowflake-Labs/sansshell/services/util/validator"
+	"github.com/google/subcommands"
+	"os"
+)
+
+// TCPCheckCmd cli adapter for execution infrastructure implementation of [subcommands.Command] interface
+type TCPCheckCmd struct {
+	host    string
+	port    int
+	timeout uint
+}
+
+func (*TCPCheckCmd) Name() string { return "tcp-check" }
+func (*TCPCheckCmd) Synopsis() string {
+	return "Check tcp connectivity from remote machine to specified server"
+}
+func (*TCPCheckCmd) Usage() string {
+	return `tcp-check --host <host> --port <port>:
+    Makes tcp connectivity check from the remote machine to specified server.
+`
+}
+
+func (p *TCPCheckCmd) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&p.host, "host", "", "Host to check connectivity from remote machine")
+	f.IntVar(&p.port, "port", 8080, "Port to check connectivity from remote machine")
```

Comment:
Uintvar?

Also, maybe make it a default of 0 and fail if set to 0? A default of 8080 doesn't seem useful.

If an arg is always required, we often get it from f.Args() instead of a flag value.

Reply:
> Uintvar?

Agreed

> Also, maybe make it a default of 0 and fail if set to 0? A default of 8080 doesn't seem useful.

Agreed

> If an arg is always required, we often get it from f.Args() instead of a flag value.

I uses flags because IMAO it gives more transparency when you read/share/use commands. It just came to my mind I could use `<host>:<port>` string as position argument and it would be much easier for end users to use

Reply:
Changed